cmake/extra_modules.cmake

```diff
@@ -163,7 +163,28 @@ endif()\n")
             endif()
             if(BUILD_${module_name})
                 message(STATUS "Register ${module_name} to be built in build-modules/${module_name}")
+                set(_OLD_CXX_FLAGS ${CMAKE_CXX_FLAGS})
+                set(_OLD_C_FLAGS ${CMAKE_C_FLAGS})
+                set(_OLD_LINKER_FLAGS ${CMAKE_EXE_LINKER_FLAGS})
+                set(_OLD_WARNING_AS_ERROE ${CMAKE_COMPILE_WARNING_AS_ERROR})
+                # temporarily remove CMAKE_COMPILE_WARNING_AS_ERROR for nvidia plugin and npu plugin
+                if ((${module_name} STREQUAL "nvidia_plugin" OR ${module_name} STREQUAL "npu") AND
+                    CMAKE_COMPILE_WARNING_AS_ERROR AND WIN32)
+                    if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+                        if(CMAKE_VERSION VERSION_LESS 3.24)
+                            string(REPLACE "/WX" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
+                            string(REPLACE "/WX" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
```

Comment:
looks like we can just set `/WX:NO`
https://learn.microsoft.com/en-us/cpp/build/reference/wx-treat-linker-warnings-as-errors?view=msvc-170

Reply:
I don't know why I get an error 
`cl : Command line error D8021 : invalid numeric argument '/WX:NO'` 
when I set /WX:NO 
`set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /WX:NO")`
`set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /WX:NO")`

Reply:
> Since we are in cmake function scope, we don't need to restore variables values, because function has its own scope of variables
Yes, it's true, but if I don't restore variables values here, CMAKE_COMPILE_WARNING_AS_ERROR=OFF will be used for all extra modules. Do I need to use it for all modules or is it only used for NV plugin and NPU plugin?

Reply:
> Since we are in cmake `function` scope, we don't need to restore variables values, because `function` has its own scope of variables
Yes, it's true, but if I don't restore variables values here, CMAKE_COMPILE_WARNING_AS_ERROR=OFF will be used for all extra modules. Do I need to use it for all modules or is it only used for NV plugin and NPU plugin?

Reply:
> Since we are in cmake `function` scope, we don't need to restore variables values, because `function` has its own scope of variables

Yes, it's true, but if I don't restore variables values here, CMAKE_COMPILE_WARNING_AS_ERROR=OFF will be used for all extra modules. Do I need to use it for all modules or is it only used for NV plugin and NPU plugin?

Reply:
we have auto generated file with initial header here https://github.com/openvinotoolkit/openvino/blob/046883335a2dcd38e6d634d6ce1274aaa960b994/cmake/extra_modules.cmake#L118

maybe we can extend this file with extra option? Like `set(CMAKE_COMPILE_WARNING_AS_ERROR OFF)`

Reply:
> we have auto generated file with initial header here
> 
> https://github.com/openvinotoolkit/openvino/blob/046883335a2dcd38e6d634d6ce1274aaa960b994/cmake/extra_modules.cmake#L118
> 
> maybe we can extend this file with extra option? Like `set(CMAKE_COMPILE_WARNING_AS_ERROR OFF)`

According to my test, for the nvidia plugin, the link error occurs in this statement which is at the beginning of CMakeLists.txt
`project(OpenVINONVIDIAGpuPlugin CXX CUDA)`
https://github.com/openvinotoolkit/openvino_contrib/blob/9cf3f168579ba558a4c5d1bdf83ce8caa2573168/modules/nvidia_plugin/CMakeLists.txt#L6
while find_package is called after in CMakeList.txt, so I think this method should have no effect on NV plugin. 
the linker error (https://jira.devtools.intel.com/browse/CVS-134705) of NV plugin like this:
` LINK : warning LNK4098: defaultlib 'LIBCMT' conflicts with use of other libs; use /NODEFAULTLIB:library`
`LINK : error LNK1218: warning treated as error; no output file generated`

Reply:
updated, disable warning as error for all extra module on windows.